### PR TITLE
translations for ES and PT-BR locales

### PR DESCRIPTION
**Pull request by LagoonProject**
This PR adds translations in Portuguese and Spanish for `es.ts` and `pt-br.ts` for the "names" key (`locale.names`). 
- No other file have been edited.
- No entries have been deleted although git says there is some deletions, it's mostly due to line width in my Prettier settings.

This should add :
- 213 entries in `pt-br.ts`
- 109 entries in `es.ts`

All the keys have been taken from existing files in romcal: 
```
- general-calendar > proper-of-saints.ts
- particular-calendars > brazil.ts
- particular-calendars > spain.ts
- particular-calendars > portugal.ts
- particular-calendars > argentina.ts
- particular-calendars > mexico.ts
```

There is no new keys added.

**Notes :**

I tried to keep things consistant with the existing entries.

So, for `es.ts` the titles (vocation) start with upper case like so : 
`cajetan_of_thiene_priest: 'San Cayetano de Thiene, Sacerdote',`

But for `pt-br.ts` the titles start with lower case like so : 
`aloysius_gonzaga_religious: 'São Luís Gonzaga, religioso',`

Most of the titles come from `en.ts`.

Sometimes there are several acceptable names for the same saint, in which case I put the alternative name between parenthesis when I could. However some alternative names might be missing for some saints.

**Sources :**

For es.ts:
- Mostly these catholic websites in spanish : 
http://es.catholic.net/op/santoral/santoral.html
https://www.corazones.org/liturgia/santos/a_santos_vidas_2.htm

- also occasionally
https://www.vaticannews.va/es/santos.html
https://es.wikipedia.org/

For pt-br : 
- Mostly these catholic websites in portuguese : 
https://www.acidigital.com/santoral/index.html
https://santo.cancaonova.com/santos/
https://franciscanos.org.br/vidacrista/calendario/#gsc.tab=0

- also occasionally:
https://www.vaticannews.va/pt/santo-do-dia.html
https://pt.wikipedia.org/

### CODE REVIEW

**Review by emagnier — APPROVED** [review]
Hello @LagoonProject! Thank you very much for your contribution and the precision of your work!! This is very appreciated! 🙏 

Thank you for keeping consistencies. About titles starting with upper or lowercase, this is something that could be adjusted for the whole locale file, if necessary (if there are language conventions). But this is something that could be done later in a future PR.

I leave a little time to @tukusejssirs if he wants to add something, and I will merge.

**Comment by tukusejssirs** [comment]
Let me have a more thorough look at this PR tomorrow, but the first question that comes to my mind is why did you, @LagoonProject, create a single PR for updates of two locales? It seems like @emagnier is okay with that, but I personally would prefer splitting the PR.

**Comment by emagnier** [comment]
@tukusejssirs no props here for the single PR/commit (it's enough scoped on these 2 locales `es` and `pt-BR` only).

**Comment by LagoonProject** [comment]
Hello, thank you guys for your feedback.

> Let me have a more thorough look at this PR tomorrow, but the first question that comes to my mind is why did you, @LagoonProject, create a single PR for updates of two locales? It seems like @emagnier is okay with that, but I personally would prefer splitting the PR.

 @tukusejssirs, sorry about that, first time I do a PR ever lol.

I can make another PR if you want. 

So I would change pt-br.ts make a pull request.
And then change es.ts and make another pull request ?

Is that the process? 

Thanks.

File: lib/locales/es.ts [hunk 1]
```diff
@@ -187,55 +187,67 @@ export const locale: Locale = {
...
     blessed_martyrs_of_paris: 'Beatos Mártires de París en la Revolución Francesa',
     bonaventure_of_bagnoregio_bishop: 'San Buenaventura, Obispo y Doctor',
     boniface_of_mainz_bishop: 'San Bonifacio, Obispo y Mártir',
+    bridget_of_sweden_religious: 'Santa Brígida de Suecia',
     brigid_of_kildare_virgin: 'Santa Brígida de Kildare, Abadesa',
     brigid_of_kildare_virgin_copatroness_of_ireland: 'Santa Brígida de Kildare, Vírgen y Patrona Secundaria de Irlanda',
     bruno_of_cologne_priest: 'San Bruno, Sacerdote',
+    cajetan_of_thiene_priest: 'San Cayetano de Thiene, Sacerdote',
+    callistus_i_pope: 'San Calixto I, Papa y Mártir',
+    camillus_de_lellis_priest: 'San Camilo de Lelis (Lellis), Sacerdote',
```
Comment (tukusejssirs) [inline lib/locales/es.ts]: Based on [this](https://liturgiapapal.org/attachments/article/1137/Calendario%2022-23.pdf#page=90) and [this](https://www.conferenciaepiscopal.es/wp-content/uploads/2022/11/CLP-y-salmos-responsoriales-2022-2023_internet.pdf#page=246) ordos, I suggest to change this.

Note that I haven’t checked all celebration names, that could be a task for you if you want.

```suggestion
    camillus_de_lellis_priest: 'San Camilo de Lelis, presbítero',
```

**Review by tukusejssirs — CHANGES_REQUESTED** [review]
> sorry about that, first time I do a PR ever lol.
>
> I can make another PR if you want.

Well, I personally would prefer a PR of a single scope, like:
- update a single locale;
- update a single calendar.

That said, I know it is not always possible and even I have created many big, multi-scoped PRs.

However, @emagnier is okay that this PR updates two locales, so you don’t need to split the PR.

---

> All the keys have been taken from existing files in romcal

I is nice you had a look into those files, but all strings are defined in `lib/locales/en.ts`. While I understand you wanted to localise primarily all l10n strings used in any and all Spanish/Portuguese-speaking countries, you might go even further and try and localise all strings used by romcal. I did the same for `cs.ts`, `la.ts`, `sk.ts` (although some, recently added strings might be missing ATM). That way, one could generate any calendar in Spanish/Portuguese locale, e.g. Chinese calendar in Spanish or Slovak Calendar in Portuguese. I understand that it is not that important, but I consider it quite important.

---

> So, for es.ts the titles (vocation) start with upper case like so :
> cajetan_of_thiene_priest: 'San Cayetano de Thiene, Sacerdote',
>
> But for pt-br.ts the titles start with lower case like so :
> aloysius_gonzaga_religious: 'São Luís Gonzaga, religioso',

Well, I believe we have no access to official liturgical practice in Spanish language. We try to follow the localisation (and other) rules based on the official liturgical style used in official liturgical books, esp in the Roman Missal. Therefore, the question is if you have access to the Spanish/Portuguese translation of the Roman Missal. I would be very grateful if you could provide us photos of the General Roman Calendar from those translations (or any other than French, English, Slovak, Hungarian, which I have access to) of the Roman Missal. I want to create locale-specific l10n rules that we would require any contributor to follow.

FYI: I divide the sources by type this way:

- primary sources (the Roman Missal, the Liturgy of the Hours, decrees issued by the Holy See, Acta Apostolicæ Sedis);
- secondary sources (other documents issued by the Holy See or local episcopal conferences, Ordos / liturgical calendars officially published by liturgical commisions of local episcopal conferences);
- tertiary sources (any other sources, like Wikipedia, other Internet sources).

> Sometimes there are several acceptable names for the same saint, in which case I put the alternative name between parenthesis when I could. However some alternative names might be missing for some saints.

I am against any deviation from the official celebration names, therefore I suggest to remove those _alternative names_. For celebration names (incl names of saints and blesseds) that are not celebrated in a particular locale, try to search the Internet for the best possible translation (or do the translation yourself), but still: pick one, best possible translation. Yes, I understand that some locales use some alternative names (or extra information) for some saints/blesseds, but I suggest to try not to use these. Keep surnames (or ‘place names’ like `of Avila` in `St Teresa of Jesus of Avila`) to make the persons (esp those of religious persons) easily identifiable and less ambiguous.

---

I suggest to change this PR to WIP/draft (use `Convert to draft` button in the sidebar) until you think you’re done with it, then convert it back and ask @emagnier and me to review your changes. :pray:

I haven’t actually reviewed all changed strings, but I believe there are many stuff that should be changed if we want to make it good as possible. @LagoonProject, are you willing to take up the task? It might take some time, be prepared. We will be very thankful if you could work on the improvement of these locales. We need many contributors and you could be really helpful in these locales (even a consultant is better than nothing).

If you are not willing to take up the task, I could do it based on the ordos found on the Internet, but I need to find some more time, but in any case after #393.

---

Tasks:

- [ ] change the strings based on the Roman Missal (if you get access to it) or at least based on [this](https://liturgiapapal.org/attachments/article/1137/Calendario%2022-23.pdf) and/or [this](https://www.conferenciaepiscopal.es/wp-content/uploads/2022/11/CLP-y-salmos-responsoriales-2022-2023_internet.pdf) Spanish ordos:
  - I could not find ordo in Portuguese (maybe it is called [_agenda diretório litúrgico_](https://livrariadm.pt/loja/papelaria/agendas-calendarios/agenda-diretorio-liturgico-2023/) or [_diretório da liturgia_](https://loja.paulus.com.br/diretorio-da-liturgia-da-igreja-no-brasil-2023-versao-de-bolso/p));
- [ ] consider localising all strings from `en.ts` in `es.ts` and `pt-br.ts`:
  - if you do not want to this in this PR (or at all), just disregard this task; 
- [ ] run `npm run lint:fix && npm run prettier:fix` after you do the changes:
  - I understand you have different style preferences than romcal uses (even I prefer some other styles over those used in romcal), but I think we should try to keep romcal style across all files in its codebase;
  - you should do this when you think you are done with the PR and want to ask us to review your changes.

**Comment by LagoonProject** [comment]
Hi @tukusejssirs, thanks for your feedback. Here's a quick response :

I have read all your points thoroughly, it makes perfect sense and I will update you on them in the following days.


> change the strings based on the Roman Missal (if you get access to it) or at least based on [this](https://liturgiapapal.org/attachments/article/1137/Calendario%2022-23.pdf) and/or [this](https://www.conferenciaepiscopal.es/wp-content/uploads/2022/11/CLP-y-salmos-responsoriales-2022-2023_internet.pdf) Spanish ordos:

Sure. I'm going to update this PR based on your comments in the following days. 

>  consider localising all strings from en.ts in es.ts and pt-br.ts:
> if you do not want to this in this PR (or at all), just disregard this task;

I can do that, but it's going to take some time because en.ts has a lot of entries. I will need to work on it piece by piece. I can offer to work on it in a future PR if you're okay with it. Meaning after this current PR is approved I will start working on the remaining entries in en.ts. I will probably need to cut it into several PRs to get through it.

As a side note a lot of entries in portuguese and spanish are currently falling back to english. That's why I focused on those entries in order to improve the user experience.

> run npm run lint:fix && npm run prettier:fix after you do the changes:

Sure, I will do it next time.

Thanks again for your review, I will provide an update hopefully pretty soon.

**Comment by miguelms95** [comment]
Hi @LagoonProject, i'm interested in this PR to be approved, for the es.ts changes. Are you going to change the requested line? Thanks in advance!

**Comment by emagnier** [comment]
@tukusejssirs is right about translations needing to closely follow the official translations. It's a goal that we absolutely must pursue.

@LagoonProject, however, your contribution is already highly substantial and of great quality, enhancing the user experience of Romcal in Spanish and Portuguese.
So if you don't have the time to work on it in the coming days, I'm in favor of merging your PR as it is (I'll do it in the next few days if you're unable to respond before then, no worries). Let's hold onto @tukusejssirs's recommendations for a next PR.

In general, let's proceed step by step with contributions, aiming to encourage more individuals to contribute, and based on each person's available time (as long as each contribution brings value and quality).

**Comment by LagoonProject** [comment]
> @tukusejssirs is right about translations needing to closely follow the official translations. It's a goal that we absolutely must pursue.
> 
> @LagoonProject, however, your contribution is already highly substantial and of great quality, enhancing the user experience of Romcal in Spanish and Portuguese. So if you don't have the time to work on it in the coming days, I'm in favor of merging your PR as it is (I'll do it in the next few days if you're unable to respond before then, no worries). Let's hold onto @tukusejssirs's recommendations for a next PR.
> 
> In general, let's proceed step by step with contributions, aiming to encourage more individuals to contribute, and based on each person's available time (as long as each contribution brings value and quality).

Hi @emagnier, sorry for the delay in my contribution, I have been terribly busy and didn't had time to finish the updated PR as per @tukusejssirs instructions. 

However I have started it and I will try to provide a finished version in the upcoming days. Basically it should be the same scope but with more rigourous titles.

Thanks for your patience, and sorry again for the delay.

**Comment by LagoonProject** [comment]
Hi @emagnier, @tukusejssirs, I have updated the pull request with a new commit with more rigorous naming for the titles of the es and pt-br locales, I have put the titles in lowercase for es.ts as it seems to be the official way of presenting the titles.

This hopefully should avoid falling back to english for most entries in the given locale, but some review might be needed.

However, I was not able to run `npm run prettier:fix` as requested, it gave an error `[error] No files matching the pattern were found: "{.github,./*}"`

If you have any comment or need any change please let me know.

**Comment by emagnier** [comment]
Hello @LagoonProject, thank you very much for the update.

We are lucky to have a new contributor for the Spanish localization (cc @miguelms95), who recently also added some improvements (already merged into the `dev` branch). As a result, this is causing some conflicts with your PR (you both edited the same locale keys).

Are you comfortable with rebasing from the latest version of the `dev` branch and resolving the conflicts? If you haven't done this before or are not very familiar with git, I can do it for you without any problem! Just let me know.

And by doing so, we'll be able to merge your work and move forward with the Spanish improvements.

**Comment by LagoonProject** [comment]
Hi, @emagnier I tried rebasing and solving the conflicts but I'm not sure it worked. I'm not really used to doing it and find it very confusing to be honest.

Github still mentions conflicts in locales/es.ts but I don't know why.

**Comment by emagnier** [comment]
@LagoonProject, no problem at all. This operation can often be a bit tricky. I've just taken your branch and updated it.

In the future (for your upcoming contributions to this project), it's better (and a good practice) not to work directly on the `dev` branch. The `dev` branch in your fork (of the Romcal repository) should only be used to stay regularly synchronized with the `dev` branch of the main Romcal repository. Then, when you begin a new work, create a new working branch (give it a name of your choice) from your updated `dev` branch.

---

While rebasing your branch, I had to deal with some conflicts and make some choices, and especially those:

- `andrew_dung_lac_priest_and_companions_martyrs`
Kept: `Santos Andrés Dung-Lac, presbítero, y compañeros, mártires`
Instead of: `San Andrés Dung-Lac y compañeros, Mártires`

- `andrew_kim_tae_gon_priest_paul_chong_ha_sang_and_companions_martyrs`
Kept: `Santos Andrés Kim Taegŏn, presbítero, Pablo Chŏng Hasang, y compañeros, mártires`
Instead of: `San Andrés Kim Taegŏn, San Pablo Chŏng Hasang y compañeros, Mártires`

- `dedication_of_the_lateran_basilica`
Kept: `Dedicación de la basílica de Letrán`
Instead of: `Dedicación de la Basílica de San Juan de Letrán`

- `lawrence_of_rome_deacon: `
Kept: `San Lorenzo, diácono y mártir`
Instead of: `San Lorenzo de Roma, Diácono`

- `teresa_of_jesus_jornet_ibars_virgin`
Kept: `Santa Teresa de Jesús Jornet e Ibars, virgen`
Instead of: `Santa Teresa de Jesús Jornet, Virgen`

- `leo_i_the_great_pope`
Kept: `San León Magno, papa y doctor de la Iglesia`
Instead of: `San Leo el Magno, papa y doctor de la Iglesia`

Let me know if everything's is ok for you. @miguelms95, I'm also open to your feedback here, since you've also worked on these translations. If everything goes well, I'll merge the pull request.

**Comment by LagoonProject** [comment]
> In the future (for your upcoming contributions to this project), it's better (and a good practice) not to work directly on the dev branch. The dev branch in your fork (of the Romcal repository) should only be used to stay regularly synchronized with the dev branch of the main Romcal repository. Then, when you begin a new work, create a new working branch (give it a name of your choice) from your updated dev branch.

Thank you @emagnier for these informations and sorry for these hiccups in my contribution I'm still learning. I'll keep your comment in mind for next time. 

Just a quick question: when I'm finished with my second branch should I merge with 'dev'? Or should I push into my second branch? If I push into my second branch will it appear in the pull request and then you merge it yourself with 'dev'?

> While rebasing your branch, I had to deal with some conflicts and make some choices, and especially those:

Yes, the conflicts you kept are right, thanks again for your help.

**Comment by miguelms95** [comment]
> Let me know if everything's is ok for you. @miguelms95, I'm also open to your feedback here, since you've also worked on these translations. If everything goes well, I'll merge the pull request.

Hi @emagnier, I've checked with the oficial CEE (Spanish Episcopal Conference) and they don't use the special accent in spanish: Santos Andrés Kim Taegŏn, presbítero, Pablo Chŏng Hasang, y compañeros, mártires. The other ones are ok 👍🏼

I'm not 100% sure, but I suggest that the correct one would be: "Santos Andrés Kim Taegon, presbítero, Pablo Chong Hasang, y compañeros, mártires" with the normal "o".

I will check in the missal and tell you asap.

**Comment by emagnier** [comment]
> Thank you @emagnier for these informations and sorry for these hiccups in my contribution I'm still learning. I'll keep your comment in mind for next time.
>
> Just a quick question: when I'm finished with my second branch should I merge with 'dev'? Or should I push into my second branch? If I push into my second branch will it appear in the pull request and then you merge it yourself with 'dev'?

@LagoonProject, no problem, we're here to learn! 😉 

Once you have finished working on your second branch (or feature / working branch), and your changes have been merged (via a Pull Request), new commits from your working branch are then merged into the `dev` branch of the main Romcal repository. Therefore, after this, there's no need to merge your branch again into the `dev` branch of your fork. Instead, you should synchronize the `dev` branch of your fork with the `dev` branch of the main Romcal repository (to retrieve all the latest updates & commits made in the main Romcal repository).

In this PR, you've been working directly from the `dev` branch. So do not make any changes to this branch until this PR is merged. Afterward, I will show you how to synchronize this branch from your fork with the main Romcal repository.

**Comment by miguelms95** [comment]
Hi again @emagnier, I've checked the translations, and the oficial text of the CEE is: "Santos Andrés Kim Taegon, presbítero, Pablo Chong Hasang y compañeros, mártires" with the normal "o".

telegram-cloud-photo-size-4-6008127560673771196-y

**Comment by emagnier** [comment]
Perfecto 👌. I'm going to fix it ASAP and merge this PR.

Small observation: I notice in your photo that the word "San" is placed in front of each name.
On liturgical books in French (as I am French), this is also similar: in the Missal, the equivalent "San" is also placed before each name, whereas in the Lectionary, the equivalent "Santos" is used. This could possibly be something to consider for standardizing within Romcal in the future...

**Review by emagnier — APPROVED** [review]

**Comment by emagnier** [comment]
Merged !

And thank you, @LagoonProject, for your great contribution !! 🎉 

I'm not forgetting your last question, I'll take time to answer you, probably tomorrow evening.

**Comment by LagoonProject** [comment]
> > Thank you @emagnier for these informations and sorry for these hiccups in my contribution I'm still learning. I'll keep your comment in mind for next time.
> > Just a quick question: when I'm finished with my second branch should I merge with 'dev'? Or should I push into my second branch? If I push into my second branch will it appear in the pull request and then you merge it yourself with 'dev'?
> 
> @LagoonProject, no problem, we're here to learn! 😉
> 
> Once you have finished working on your second branch (or feature / working branch), and your changes have been merged (via a Pull Request), new commits from your working branch are then merged into the `dev` branch of the main Romcal repository. Therefore, after this, there's no need to merge your branch again into the `dev` branch of your fork. Instead, you should synchronize the `dev` branch of your fork with the `dev` branch of the main Romcal repository (to retrieve all the latest updates & commits made in the main Romcal repository).
> 
> In this PR, you've been working directly from the `dev` branch. So do not make any changes to this branch until this PR is merged. Afterward, I will show you how to synchronize this branch from your fork with the main Romcal repository.

Ok I understand now, thanks for the clarification. I will not use the 'dev' branch directly anymore. Here's a screenshot of my repository by the way, there's a "sync fork" button on the right. I have not used it yet.

capture_romcal

**Comment by emagnier** [comment]
Yes exactly, that's the simplest way offered by GitHub. Now that your pull request (and your work) have been merged, you can proceed to synchronize the `dev` branch of your fork 👍

**Comment by LagoonProject** [comment]
> Yes exactly, that's the simplest way offered by GitHub. Now that your pull request (and your work) have been merged, you can proceed to synchronize the `dev` branch of your fork 👍

Got it. Thanks a lot.

**Pull request merged**